### ISSUE TITLE
Need improved invoice generation

### ISSUE THREAD

**Issue by adis1987** [kimai/kimai#703]
I want to generate invoice **only** for the projects that I have worked with in the given time span. As it is now, only tasks and expenses is shown. It would be better to show:

[project] [task] [time comsumed] [price]

**Comment by kevinpapst** [comment]
What about the short invoice option?
You know that you can adjust the invoice templates to your needs?

**Comment by adis1987** [comment]
Yes I know that. Short/long, they are both good in their ways. But how do I include to what project every line is refering to?

In the current templates I can find this line:
[row.desc][row;block=table:table-row; when [row.type] = timeSheet]
I want to add corresponding project there to.

Also for this:
[row.desc][row;block=table:table-row; when [row.type] = expense]
That code shows expenses, but I want to add the project number to every expensen line.

**Comment by kevinpapst** [comment]
Here is a documentation for writing invoice templates: http://www.kimai.org/documentation/developer/invoice-templates.html

Unfortunately I guess you won't be able to fetch the values you need in Kimai 0.9.3. 

I worked quite a lot on the invoice extension for the next version, so if you would try the current development version from here: https://github.com/kimai/kimai/archive/master.zip - you should be able to use the variables row.projectID and row.projectName.

You can find all available variables for the invoices in this function: https://github.com/kimai/kimai/blob/master/extensions/ki_invoice/private_func.php#L33

**Comment by adis1987** [comment]
Tested it now. That is what I was looking for. It works perfect if not using short invoice. When using short invoice, only the first project is shown; instead it should show all with comma separation.
This will be great!

**Comment by kevinpapst** [comment]
Short invoice probably doesn't make a lot of sense when using multiple projects?!? Or we should rethink the complete "short invoice concept".

I am not sure how to improve it, as I never use Kimai myself for printing short invoices. 
Do you have any recommendation? 
What is your use-case? 
Could you help me understanding how you use it and what you want to display on a short invoice? :-)

**Comment by adis1987** [comment]
As you said, I also think it is pointless to use short invoice then. But! If I use short invoice, then it would be nice to have a parameter that shows a project list, comma separated , within the given time span.

If I have 200 project, and worked on 15 projects, it is hard to remember them and manually select them in invoice menu. I then select them all and automagic I get a list only of them 15 project even if I selected 200 project.

The point with that is that in the short invoice it can then be stated:
The list below relates to projects: project1, project2 ... project15

Hope you understand. I would love to help and test.

**Comment by adis1987** [comment]
When will this version be released?

I get error "Failed rendering invoice: item '" in browser when tring to generate a invoice (odt-file) that has [row.projectComment].

**Comment by kevinpapst** [comment]
Very soon. I just need some time to incorporate last bugfixes / open pull requests.
After that - I plan to do it in the next two weeks. So the master version should be quite stable and not change much.
